app/src/test/kotlin/maru/app/MaruAppCliTest.kt

```diff
@@ -8,16 +8,152 @@
  */
 package maru.app
 
+import java.io.File
+import maru.config.MaruConfigDtoToml
+import maru.config.MaruConfigLoader.parseBeaconChainConfig
+import maru.config.MaruConfigLoader.parseConfig
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
 import picocli.CommandLine
 
 class MaruAppCliTest {
-  private val cmd = CommandLine(MaruAppCli(true))
+  companion object {
+    object NoOpMaruApp : LongRunningCloseable {
+      override fun start() = Unit
+
+      override fun stop() = Unit
+
+      override fun close() = Unit
+    }
+
+    private val maruConfigDtoToml =
+      """
+      [persistence]
+      data-path="/tmp/maru-db"
+
+      [observability]
+      port = 9090
+      jvm-metrics-enabled = true
+      prometheus-metrics-enabled = true
+
+      [api]
+      port = 8080
+
+      [payload-validator]
+      engine-api-endpoint = { endpoint = "http://localhost:8550" }
+      eth-api-endpoint = { endpoint = "http://localhost:8545" }
+
+      [syncing]
+      peer-chain-height-polling-interval = "5 seconds"
+      sync-target-selection = "Highest"
+      el-sync-status-refresh-interval = "5 seconds"
+      """.trimIndent()
+
+    private val maruConfigOverridesDtoToml =
+      """
+      [persistence]
+      data-path="./OVERRIDE/maru-db"
+      private-key-path="./OVERRIDE/maru-db/private-key"
+
+      [payload-validator]
+      engine-api-endpoint = { endpoint = "http://OVEERRIDE:8550" }
+
+      [syncing]
+      peer-chain-height-polling-interval = "10 seconds"
+      """.trimIndent()
+
+    private val expectedMaruConfigDtoToml =
```

Comment:
Nit: It's not necessary to use a string here. You could have just instantiated the whole domain object instead. That way you make the assertion less dependent on the parsing logic, while keeping the copy-paste at the same level or lower, due to defaults set in the domain classes